src/commands.cpp

```diff
@@ -4011,9 +4011,14 @@ void CommandGetUserQuota::procresult()
                 break;
 
             case MAKENAMEID6('s', 'u', 'n', 't', 'i', 'l'):
+            {
                 // expiry of last active Pro plan (may be different from current one)
                 details->pro_until = client->json.getint();
-                break;
+                m_time_t now = m_time(nullptr);
+                dstime diff = (details->pro_until - now) * 10;
+                client->btsuntilexpiration.backoff(diff);
```

Comment:
Nice use of the backoff timers, however the client side clock could be ahead of the clocks on the servers by a few second, and so the timer could go off early (from the server side perspective).  I think we should allow for that, perhaps by retrying a few times with increasing backoff times, or coordinating with the app layer somehow?  Once the account is detected to have lapsed back to a free account, we can cancel the backoffs.   Or is this field changed/absent for a lapsed-back-to-free account?  thanks

Reply:
If I understand the backoff correctly, I expect it to not be triggered unless the `uq` has returned an `suntil`. Once it is armed, it should result in only one `uq` and no more (as long as the account has lapsed back to a free account and remains as free).

On the other hand, I agree about the clock's skew issue. Instead of over-complicating this case with retries and alike, perhaps we can simply add a few additional seconds to the backoff? Perhaps not ideal, but simpler and effective, IMHO.

Reply:
Yes, that will avoid the issue initially.  Also I was concerned that we could get in a situation where the timer expires, we send uq, get the `suntil`, set the timer again which expires soon or immediately, and now we're in a loop burning the user's bandwidth, and DDOSing ourselves - so long as we protect against that somehow, I think it's good.  thanks

Reply:
A possible workaround could be similar to the `cachedug` flag for `ug` requests. If the flag is set, no request is sent again when the timer expires (unless the app explicitly requests it). When the backoff is armed, we reset the flag and send the request (which I think it will have the common exponential backoffs of any request). Upon response, we set the flag. Additionally, if the `suntil` is in the past, simply forget about it to not request once and again.
@polmr would you mind to apply the changes above if you agree with the suggestion? Thanks